cmd/manager/main.go

```diff
@@ -37,173 +52,156 @@ import (
 	signjob "github.com/kubernetes-sigs/kernel-module-management/internal/sign/job"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/statusupdater"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
-
-	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
-	// to ensure that exec-entrypoint and run can make use of them.
-	_ "k8s.io/client-go/plugin/pkg/client/auth"
-
-	"k8s.io/apimachinery/pkg/runtime"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
-	"k8s.io/klog/v2"
-	"k8s.io/klog/v2/klogr"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/healthz"
-
-	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
-	"github.com/kubernetes-sigs/kernel-module-management/controllers"
 	//+kubebuilder:scaffold:imports
 )
 
 var scheme = runtime.NewScheme()
 
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
-
-	utilruntime.Must(kmmv1beta1.AddToScheme(scheme))
+	utilruntime.Must(v1beta12.AddToScheme(scheme))
 	//+kubebuilder:scaffold:scheme
 }
 
 func main() {
-	var (
-		configFile           string
-		metricsAddr          string
-		enableLeaderElection bool
-		probeAddr            string
-	)
+	logger := klogr.New().WithName("kmm")
 
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
+	ctrl.SetLogger(logger)
+
+	setupLogger := logger.WithName("setup")
+
+	var configFile string
 
 	flag.StringVar(&configFile, "config", "", "The path to the configuration file.")
 
 	klog.InitFlags(flag.CommandLine)
 
 	flag.Parse()
 
-	logger := klogr.New()
-
-	ctrl.SetLogger(logger)
-
-	setupLogger := logger.WithName("setup")
-
-	commit, err := gitCommit()
+	commit, err := cmd.GitCommit()
 	if err != nil {
 		setupLogger.Error(err, "Could not get the git commit; using <undefined>")
 		commit = "<undefined>"
 	}
 
+	managed, err := GetBoolEnv("KMM_MANAGED")
```

Comment:
Can we set this flag into the ConfigMap, somehow?

Reply:
This is much easier to override in the [`SubscriptionConfig`](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/subscription-config.md). If we use the `ConfigMap,` we need the admin to first createit with the appropriate value and then to replace the `volumeMount` entry appropriately. This looked more complex to me for now real benefit.
BTW the custom type for settings is gone in this PR.